src/aaudio/AudioStreamAAudio.cpp

```diff
@@ -264,37 +264,37 @@ Result AudioStreamAAudio::requestStop() {
     }
 }
 
-ErrorOrValue<int32_t>   AudioStreamAAudio::write(const void *buffer,
+ResultWithValue<int32_t>   AudioStreamAAudio::write(const void *buffer,
                                      int32_t numFrames,
                                      int64_t timeoutNanoseconds) {
     AAudioStream *stream = mAAudioStream.load();
     if (stream != nullptr) {
         int32_t result = mLibLoader->stream_write(mAAudioStream, buffer,
                                                   numFrames, timeoutNanoseconds);
         if (result < 0) {
-            return ErrorOrValue<int32_t>(static_cast<Result>(result));
+            return ResultWithValue<int32_t>(static_cast<Result>(result));
         } else {
-            return ErrorOrValue<int32_t>(result);
+            return ResultWithValue<int32_t>(result);
         }
     } else {
-        return ErrorOrValue<int32_t>(Result::ErrorNull);
+        return ResultWithValue<int32_t>(Result::ErrorNull);
     }
 }
 
-ErrorOrValue<int32_t>   AudioStreamAAudio::read(void *buffer,
+ResultWithValue<int32_t>   AudioStreamAAudio::read(void *buffer,
                                  int32_t numFrames,
                                  int64_t timeoutNanoseconds) {
     AAudioStream *stream = mAAudioStream.load();
     if (stream != nullptr) {
         int32_t result = mLibLoader->stream_read(mAAudioStream, buffer,
                                                  numFrames, timeoutNanoseconds);
         if (result < 0) {
-            return ErrorOrValue<int32_t>(static_cast<Result>(result));
+            return ResultWithValue<int32_t>(static_cast<Result>(result));
```

Comment:
I would prefer to have this logic in `ResultWithValue` instead.

Reply:
Mikhail says we could have a kind of factory class that constructed a ResultWithValue based on sign.
Something like:
    return ResultWithValue::createBasedOnSign(result);

Reply:
Have added, also works nicely.